test/e2e/apimachinery/watch.go

```diff
@@ -33,158 +31,258 @@ import (
+			setConfigMapData(cm, "mutation", "2")
+		})
+		Expect(err).NotTo(HaveOccurred())
+
+		By("deleting the configmap")
+		err = c.CoreV1().ConfigMaps(ns).Delete(testConfigMap.GetName(), nil)
+		Expect(err).NotTo(HaveOccurred())
+
+		By("creating a watch on configmaps from the resource version returned by the first update")
+		testWatch, err := watchConfigMaps(f, testConfigMapFirstUpdate.ObjectMeta.ResourceVersion, fromResourceVersionLabelValue)
+		Expect(err).NotTo(HaveOccurred())
+
+		By("Expecting to observe notifications for all changes to the configmap after the first update")
+		expectEvent(testWatch, watch.Modified, testConfigMapSecondUpdate)
+		expectEvent(testWatch, watch.Deleted, nil)
+	})
+
+	/*
+		    Testname: watch-configmaps-closed-and-restarted
+		    Description: Ensure that a watch can be reopened from the last resource version
+			observed by the previous watch, and it will continue delivering notifications from
+			that point in time.
+	*/
+	framework.ConformanceIt("should be able to restart watching from the last resource version observed by the previous watch", func() {
+		c := f.ClientSet
+		ns := f.Namespace.Name
+
+		testConfigMap := &v1.ConfigMap{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "e2e-watch-test-watch-closed",
+				Labels: map[string]string{
+					watchConfigMapLabelKey: watchRestartedLabelValue,
+				},
+			},
+		}
+
+		By("creating a watch on configmaps")
+		testWatchBroken, err := watchConfigMaps(f, "", watchRestartedLabelValue)
+		Expect(err).NotTo(HaveOccurred())
+
+		By("creating a new configmap")
+		testConfigMap, err = c.CoreV1().ConfigMaps(ns).Create(testConfigMap)
+		Expect(err).NotTo(HaveOccurred())
+
+		By("modifying the configmap once")
+		testConfigMapFirstUpdate, err := updateConfigMap(c, ns, testConfigMap.GetName(), func(cm *v1.ConfigMap) {
+			setConfigMapData(cm, "mutation", "1")
+		})
+		Expect(err).NotTo(HaveOccurred())
+
+		By("closing the watch once it receives two notifications")
+		expectEvent(testWatchBroken, watch.Added, testConfigMap)
+		lastEvent, ok := waitForEvent(testWatchBroken, watch.Modified, nil, 1*time.Minute)
+		if !ok {
+			framework.Failf("Timed out waiting for second watch notification")
+		}
+		testWatchBroken.Stop()
+
+		By("creating a new watch on configmaps from the last resource version observed by the first watch")
+		lastEventConfigMap, ok := lastEvent.Object.(*v1.ConfigMap)
+		if !ok {
+			framework.Failf("Expected notfication to refer to a configmap but got: %v", lastEvent)
+		}
+		testWatchRestarted, err := watchConfigMaps(f, lastEventConfigMap.ObjectMeta.ResourceVersion, watchRestartedLabelValue)
+		Expect(err).NotTo(HaveOccurred())
+
+		By("Expecting not to observe duplicate notifications for changes already observed by the first watch")
+		expectNoEvent(testWatchRestarted, watch.Added, testConfigMap)
+		expectNoEvent(testWatchRestarted, watch.Modified, testConfigMapFirstUpdate)
+
+		By("modifying the configmap a second time")
```

Comment:
Please move that higher.

What I would like to ensure is that even if something happened in the meantime (i.e. after watch was stopped but before it was restarted), an event for that will still be delivered.

Reply:
Done, for some reason github moved your comment to a different test